src/Razor/src/Microsoft.VisualStudio.Editor.Razor/DefaultVisualStudioDocumentTracker.cs

```diff
@@ -113,15 +90,15 @@ internal class DefaultVisualStudioDocumentTracker : VisualStudioDocumentTracker
         public override Project Project =>
             _projectSnapshot.WorkspaceProject == null ?
             null :
-            _workspace.CurrentSolution.GetProject(_projectSnapshot.WorkspaceProject.Id);
+            _workspaceState.Workspace.CurrentSolution.GetProject(_projectSnapshot.WorkspaceProject.Id);
 
         internal override ProjectSnapshot ProjectSnapshot => _projectSnapshot;
 
         public override ITextBuffer TextBuffer => _textBuffer;
 
         public override IReadOnlyList<ITextView> TextViews => _textViews;
 
-        public override Workspace Workspace => _workspace;
+        public override Workspace Workspace => _workspaceState.Workspace;
```

Comment:
This seems a little crazy to me. If we don't have a snapshot, can we just assume there is no workspace project?